### PR TITLE
Doc Generation, Doc Publishing, Release tagging

### PR DESCRIPTION
**Pull request by danieljurek**
@vhvb1989 -- I'm using the old version of docs generation with the Python script for now. If we have cmake commands that can generate docs I can incorporate those!

### CODE REVIEW

**Comment by danieljurek** [comment]
This requires a merge in `azure-sdk-build-tools` before its unified pipeline builds will succeed

File: eng/docs/index/assets/docfx.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      "templates/matthews"
+    ],
+    "postProcessors": [],
+    "markdownEngineName": "markdig",
+    "noLangKeyword": false,
+    "keepFileLink": false,
+    "cleanupCacheHistory": false,
+    "disableGitFeatures": false,
+    "globalMetadata": {
+      "_appTitle": "Azure SDK for Net",
```
Comment (vhvb1989) [inline eng/docs/index/assets/docfx.json]: do we want this to be for C instead of Net?

File: eng/docs/index/assets/docgen.csproj [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (vhvb1989) [inline eng/docs/index/assets/docgen.csproj]: do we need to use this as Project?

**Comment by danieljurek** [comment]
> Added some questions about why we have .Net and Nugget references
> and why we are installing with chocolatey
> other than that, this looks good to merge

For chocolatey vs. downloading the package. We wanted to pin to a particular version. Chocolatey does have this version so I've opened an issue to track this: https://github.com/Azure/azure-sdk/issues/1067

**Pull request merged**